### ISSUE TITLE
Recovery token without user_id

### ISSUE THREAD

**Issue by klikin** [dektrium/yii2-user#926]
Hi:

Is there anyway to override the creation of the recovery URL to NOT show the user_id ?

I think it's not right to do this. I know how to change it in the code, but I wonder if there's any way to override in a cleaner way.

Thank you

**Comment by klikin** [comment]
Basically my change is:

yii2-user/models/Token.php:69
- return Url::to([$route, 'id' => $this->user_id, 'code' => $this->code], true);
+ return Url::to([$route, 'code' => $this->code], true);
and

yii2-user/controllers/RecoveryController.php:150
- $token = $this->finder->findToken(['user_id' => $id, 'code' => $code, 'type' => Token::TYPE_RECOVERY])->one();
+ $token = $this->finder->findToken(['code' => $code, 'type' => Token::TYPE_RECOVERY])->one();

(Tested and recovery process still working)

With this, user_id is not sent and is not needed to search the "code".

I think showing user_id gives information sometimes you don't want to give ;-)

Is it possible to override this cleaner without changing vendor files ? ;-)

**Comment by SamMousa** [comment]
I agree, this should be changed. We are creating a fork as we speak.
Would you care to submit a PR there? @klikin

**Comment by klikin** [comment]
I'm going to do it with my personal user, this is company account ;-)

**Comment by adantart** [comment]
Done it ! ;-)
